frontend/cypress/integration/datasets.spec.js

```diff
@@ -111,6 +112,78 @@ describe('The Datasets Pages', () => {
       .children()
       .eq(1)
-      .contains('Countries Being Left Behind');
+      .not()
+      .contains('Select Data Source');
+  });
+
+  it('views data in datasets', () => {
+    cy.fixture('datasets').then((datasets) => {
+      cy.intercept('api/datasets/mine/', datasets);
+    });
+    cy.fixture('datasetTableData').then((data) => {
+      cy.intercept('api/dataset/data/4', data);
+    });
+
+    // View dataset data in tabular form and export it as csv
+    cy.visit('/');
+    cy.get('.dataset-row')
+      .eq(3)
+      .then(($datasetRow) => {
+        cy.wrap($datasetRow)
+          .contains('View Data')
+          .click({ force: true })
+          .then(() => {
+            cy.get('[data-testid="dataset-table-body"]').children().should('have.length', 10);
+            cy.get('[data-testid="dataset-data-export-form"]').then(($form) => {
+              cy.wrap($form).should('have.attr', 'method', 'POST');
+              cy.wrap($form).should('have.attr', 'action', '/api/export/4/');
+            });
+            cy.get('[data-testid="dataset-export-button"]')
+              .should('be.visible')
+              .should('have.attr', 'type', 'submit');
+          });
+      });
+  });
+
+  it('freezes and deletes a dataset', () => {
+    cy.visit('/');
+    cy.get('.dataset-row').eq(0).contains('Versions').click({ force: true });
+    cy.get('[data-testid="dataset-freeze-button"]').click();
+    cy.get('[data-testid="dataset-frozen-data-description"]')
+      .should('be.visible')
+      .type('Test dataset freeze');
+    cy.get('[data-testid="dataset-frozen-data-save-button"]').should('be.visible').click();
+    cy.get('[data-testid="dataset-frozen-data-refresh-button"]').first().click({ force: true });
+    cy.get('[data-testid="dataset-frozen-data-status"]').contains('Completed');
+
+    // Download frozen dataset
+    cy.getAccessToken().then((token) => {
+      if (token) {
+        const options = {
+          url: `${Cypress.config('baseUrl')}/api/savedquerysets/`,
+          headers: {
+            Authorization: `token ${token.replaceAll('"', '')}`,
+          },
+        };
+        cy.request(options).then((response) => {
+          const currentDataDatasetId = Math.max(...response.body.map((data) => Number(data.id)));
+          const currentFrozenDataset = response.body.find(
+            (item) => Number(item.id) === currentDataDatasetId,
+          );
+          cy.get('[data-testid="frozen-dataset-download-button"]')
+            .eq(0)
+            .should('not.be.visible')
+            .should(
+              'have.attr',
+              'href',
+              `/api/tables/download/${currentFrozenDataset.saved_query_db_table}/dataset/`,
+            );
+        });
+      }
+    });
+
+    // Delete frozen dataset
+    cy.get('[data-testid="frozen-dataset-delete-button"]').first().dblclick({ force: true });
+    cy.get('.p-0').contains('Test dataset freeze').should('not.exist');
```

Comment:
This will fail when there are more than one records with duplicate names... count rows instead.